syntax = "proto3";

package feed;

service FeedService {
  rpc GetFeed(GetFeedRequest) returns (GetFeedResponse) {}
  rpc GetArticleById(GetArticleByIdRequest) returns (GetArticleByIdResponse) {}
}

message Author {
  int32 id = 1;
  string name = 2;
}

message Article {
  int32 id = 1;
  string title = 2;
  Author author = 3;
}

message Feed {
  repeated Article articles = 1;
}

message GetFeedRequest {}

message GetFeedResponse {
  Feed feed = 1;
}

message GetArticleByIdRequest {
  int32 id = 1;
}

message GetArticleByIdResponse {
  Article article = 1;
}